source/adios2/core/IO.tcc

```diff
@@ -103,15 +103,12 @@ Attribute<T> &IO::DefineAttribute(const std::string &name, const T &value,
                                   const std::string separator)
 {
     TAU_SCOPED_TIMER("IO::DefineAttribute");
-    if (m_DebugMode)
+    if (!variableName.empty() && InquireVariableType(variableName).empty())
```

Comment:
Well, this is not free, but it's good to raise an exception anyway

Reply:
I mean, this is just some cost when defining attributes which might happen thousands of times but it will not affect IO performance. So I agree with removing the test here.

Reply:
Agreed.